src/erlcloud_ddb2.erl

```diff
@@ -2792,15 +2830,28 @@ put_item(Table, Item, Opts) ->
 -spec put_item(table_name(), in_item(), put_item_opts(), aws_config()) -> put_item_return().
 put_item(Table, Item, Opts, Config) ->
     {AwsOpts, DdbOpts} = opts(put_item_opts(), Opts),
+    IsIdempotent = do_put_item_opts_keep_idempotency(Opts),
     Return = erlcloud_ddb_impl:request(
                Config,
                "DynamoDB_20120810.PutItem",
                [{<<"TableName">>, Table},
                 {<<"Item">>, dynamize_item(Item)}]
-               ++ AwsOpts),
+               ++ AwsOpts,
+               [{idempotent, IsIdempotent}]),
     out(Return, fun(Json, UOpts) -> undynamize_record(put_item_record(), Json, UOpts) end, DdbOpts, 
         #ddb2_put_item.attributes, {ok, []}).
 
+-spec do_put_item_opts_keep_idempotency(put_item_opts()) -> boolean().
+do_put_item_opts_keep_idempotency(Opts) ->
+    lists:all(fun does_put_item_opt_keep_idempotency/1, Opts).
+
+-spec does_put_item_opt_keep_idempotency(put_item_opt()) -> boolean().
+does_put_item_opt_keep_idempotency({OptName, _}) ->
+    % https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html
+    not lists:member(OptName, [condition_expression,
```

Comment:
again, here, it seems like conditions make a request more likely to be idempotent.

Reply:
Same rationale as for the `Expected` condition I previously commented :-)